### PR TITLE
Added support for byte range curl download(RDK-24363).

### PR DESCRIPTION
**Pull request by ssureshmsc**
CDN requires the curl download request to be less range or equal to (7 * 47 * 4096). So added support for multiple call call with chunk request instead of single bigger download.

### CODE REVIEW

File: src/rtFileDownloader.cpp [hunk 1]
```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
...
+}
+
+bool rtFileDownloader::downloadByteRangeFromNetwork(rtFileDownloadRequest* downloadRequest)
+{
+    CURL *curl_handle = NULL;
+    CURLcode res = CURLE_OK;
+    char errorBuffer[CURL_ERROR_SIZE];
+    int multipleIntervals = 1;
+    size_t startRange = 0;
+    std::string byteRange("NULL");
```
Comment (mfiess) [inline src/rtFileDownloader.cpp]: Please remove std::string usage.  Please use rtString instead.

File: src/rtFileDownloader.cpp [hunk 2]
```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
...
+      if(res != CURLE_OK)
+        break;
+
+      if(iLoop == 0)
+      {
+        curl_easy_setopt(curl_handle, CURLOPT_HEADERFUNCTION, NULL);
+        curl_easy_setopt(curl_handle, CURLOPT_HEADERDATA, NULL);
+
+        // Parsing total filesize from Content-Range.
+        std::string httpHeaderStr(reinterpret_cast< char const* >((char *)chunk.headerBuffer));
```
Comment (mfiess) [inline src/rtFileDownloader.cpp]: Please remove std::string usage. Please use rtString instead.

File: src/rtFileDownloader.cpp [hunk 3]
```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
...
+        size_t findContentRange = httpHeaderStr.find("Content-Range:");
+        if(findContentRange != std::string::npos)
+        {
+          std::string strContentRange = httpHeaderStr.substr(findContentRange+1);
```
Comment (mfiess) [inline src/rtFileDownloader.cpp]: Please remove std::string usage. Please use rtString instead.

File: src/rtFileDownloader.cpp [hunk 4]
```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
...
+          size_t findRange = strContentRange.find("/");
+
+          if(findRange != std::string::npos)
+          {
+            std::string substrRange = strContentRange.substr(findRange+1);
```
Comment (mfiess) [inline src/rtFileDownloader.cpp]: Please remove std::string usage. Please use rtString instead.

File: src/rtFileDownloader.cpp [hunk 5]
```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
...
+            std::string strRange = substrRange.substr(0, substrRange.find("\n")-1);
```
Comment (mfiess) [inline src/rtFileDownloader.cpp]: Please remove std::string usage. Please use rtString instead.

File: src/rtFileDownloader.cpp (hunk 2 above)
Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

File: src/rtFileDownloader.cpp (hunk 1 above)
Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

File: src/rtFileDownloader.cpp (hunk 2 above)
Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

File: src/rtFileDownloader.cpp (hunk 4 above)
Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

File: src/rtFileDownloader.cpp (hunk 5 above)
Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

**Comment by ssureshmsc** [comment]
Changed to rtString and pushed again.

File: src/rtFileDownloader.cpp (hunk 3 above)
Comment (ssureshmsc) [reply src/rtFileDownloader.cpp]: Changed to rtString and pushed again.

**Pull request merged**